tlatools/org.lamport.tlatools/test/tlc2/model/ParameterizedInstantiation.java

```diff
@@ -0,0 +1,90 @@
+package tlc2.model;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+
+import org.junit.Test;
+
+import tlc2.output.EC;
+import tlc2.tool.liveness.ModelCheckerTestCase;
+import util.TLAConstants;
+
+/**
+ * These are test cases related to the investigation of problems involving parameterized instantiation, undertaken starting
+ * 	in April, 2020.
+ */
+public class ParameterizedInstantiation extends ModelCheckerTestCase {
+	private static final String TEST_MODEL_FOLDER_NAME = "ParameterizedInstantiation";
+	private static final String SPEC_A_CONFIG = "SPECIFICATION\nSpecA\n";
+	private static final String SPEC_B_CONFIG = "SPECIFICATION\nSpecB\n";
+	private static final String INIT_NEXT_CONFIG = "INIT\nInit\nNEXT\nNext\n";
+	private static final String PARENT_DIR = BASE_PATH + File.separator + TEST_MODEL_FOLDER_NAME;
+	private static final File CONFIG_FILE = new File(PARENT_DIR, TLAConstants.Files.MODEL_CHECK_CONFIG_FILE);
+	
+	
+	private int testNumber;
+	public ParameterizedInstantiation() {
+		super(TLAConstants.Files.MODEL_CHECK_FILE_BASENAME, TEST_MODEL_FOLDER_NAME);
+		
+		testNumber = 0;
+	}
+
+	@Override
+	protected void beforeSetUp() {
+		try {
+			switch(testNumber) {
+				case 0:
+					writeConfigFile(SPEC_A_CONFIG);
+					break;
+				case 1:
+					writeConfigFile(SPEC_B_CONFIG);
+					break;
+				case 2:
+					writeConfigFile(INIT_NEXT_CONFIG);
+					break;
+			}
+
+			// we need to wait at least a second or else we risk having TLC fail because the states directory already exists
+			Thread.sleep(1100);
```

Comment:
Is the test running with `-Dutil.FileUtil.milliseconds=true`, which should take care of this issue?

https://github.com/tlaplus/tlaplus/blob/4111431a169b96d0b2eb850fd6fb4416cf5af4ff/tlatools/org.lamport.tlatools/src/util/FileUtil.java#L242-L249

Reply:
Ooh - perhaps. I was developing the unit test in Eclipse and running it from Eclipse - so i didn't know about that argument. Will ensure it works in Eclipse and command line.

Reply:
Can you amend the error message to hint at '-Dutil.FileUtil.milliseconds=true` that TLC prints if it cannot create a states directory?

Reply:
Will do.